src/app/pages/Swap/modals/SwapSelectAsset/AllEvmChainsAssetsList.tsx

```diff
@@ -1,81 +1,164 @@
+import { EvmChain, useAllEvmChains, useEnabledEvmChains } from 'temple/front';
+import { useFavoriteTokens } from 'temple/front/use-favorite-tokens';
 import { TempleChainKind } from 'temple/types';
 
+interface ItemData {
+  searchedSlugs: string[];
+  publicKeyHash: HexString;
+  evmChains: StringRecord<EvmChain>;
+  showOnlyFavorites?: boolean;
+  showFavoritesMark?: boolean;
+  onAssetSelect: (e: React.MouseEvent, slug: string) => void;
+}
+
 interface Props {
+  activeField: SwapFieldName;
   accountEvmAddress: HexString;
+  showOnlyFavorites?: boolean;
   searchValue: string;
   onAssetSelect: (e: MouseEvent, chainSlug: string) => void;
 }
 
-export const AllEvmChainsAssetsList = memo<Props>(({ accountEvmAddress, searchValue, onAssetSelect }) => {
-  const evmTokensSlugs = useEnabledEvmAccountTokenSlugs(accountEvmAddress);
-  const enabledEvmChains = useEnabledEvmChains();
+export const AllEvmChainsAssetsList = memo<Props>(
+  ({ accountEvmAddress, activeField, showOnlyFavorites, searchValue, onAssetSelect }) => {
+    const evmTokensSlugs = useEnabledEvmAccountTokenSlugs(accountEvmAddress);
+    const enabledEvmChains = useEnabledEvmChains();
+    const { lifiTokenSlugs, isLoading } = useLifiEvmAllTokensSlugs();
 
-  const tokensSortPredicate = useEvmAccountTokensSortPredicate(accountEvmAddress);
+    const getEvmBalance = useGetEvmTokenBalanceWithDecimals(accountEvmAddress);
 
-  const enabledAssetsSlugs = useMemo(
-    () =>
-      enabledEvmChains
-        .map(chain => toChainAssetSlug(TempleChainKind.EVM, chain.chainId, EVM_TOKEN_SLUG))
-        .concat(evmTokensSlugs),
-    [enabledEvmChains, evmTokensSlugs]
-  );
+    const isEvmNonZeroBalance = useCallback(
+      (chainSlug: string) => {
+        const [, chainId, assetSlug] = parseChainAssetSlug(chainSlug);
+        return isDefined(getEvmBalance(chainId as number, assetSlug));
+      },
+      [getEvmBalance]
+    );
 
-  const enabledAssetsSlugsSorted = useMemoWithCompare(
-    () => enabledAssetsSlugs.sort(tokensSortPredicate),
-    [enabledAssetsSlugs, tokensSortPredicate]
-  );
+    const { favoriteTokens = [] } = useFavoriteTokens();
 
-  const evmChains = useAllEvmChains();
+    const filterZeroBalances = useMemo(() => activeField === 'input', [activeField]);
+    const showFavoritesMark = useMemo(() => activeField === 'output', [activeField]);
 
-  const evmMetadata = useEvmTokensMetadataRecordSelector();
+    const rawTokensSortPredicate = useEvmAccountTokensSortPredicate(accountEvmAddress, showFavoritesMark);
+    const [tokensSortPredicate] = useState(() => rawTokensSortPredicate);
 
-  const getMetadata = useCallback(
-    (chainId: number, slug: string) =>
-      slug === EVM_TOKEN_SLUG ? evmChains[chainId]?.currency : evmMetadata[chainId]?.[slug],
-    [evmChains, evmMetadata]
-  );
+    const enabledAssetsSlugs = useMemo(() => {
+      if (showOnlyFavorites) {
+        return [...favoriteTokens.filter(token => token.startsWith('evm'))];
```

Comment:
`favoriteTokens.filter` already returns a new array, so there is no need to spread it